wallet/bitcoinrpc.cpp

```diff
@@ -734,88 +822,35 @@ void ThreadRPCServer2(void* /*parg*/)
 {
     printf("ThreadRPCServer started\n");
 
-    std::string rpcUser;
-    mapArgs.get("-rpcuser", rpcUser);
-    std::string rpcPassword;
-    mapArgs.get("-rpcpassword", rpcPassword);
-
-    strRPCUserColonPass = rpcUser + ":" + rpcPassword;
-    if ((rpcPassword == "") || (rpcUser == rpcPassword)) {
-        unsigned char rand_pwd[32];
-        RAND_bytes(rand_pwd, 32);
-        string strWhatAmI = "To use nebliod";
-        if (mapArgs.exists("-server"))
-            strWhatAmI = strprintf(_("To use the %s option"), "\"-server\"");
-        else if (mapArgs.exists("-daemon"))
-            strWhatAmI = strprintf(_("To use the %s option"), "\"-daemon\"");
-        uiInterface.ThreadSafeMessageBox(
-            strprintf(
-                _("%s, you must set a rpcpassword in the configuration file:\n %s\n"
-                  "It is recommended you use the following random password:\n"
-                  "rpcuser=nebliorpc\n"
-                  "rpcpassword=%s\n"
-                  "(you do not need to remember this password)\n"
-                  "The username and password MUST NOT be the same.\n"
-                  "If the file does not exist, create it with owner-readable-only file permissions.\n"
-                  "It is also recommended to set alertnotify so you are notified of problems;\n"
-                  "for example: alertnotify=echo %%s | mail -s \"neblio Alert\" admin@foo.com\n"),
-                strWhatAmI.c_str(), GetConfigFile().string().c_str(),
-                EncodeBase58(&rand_pwd[0], &rand_pwd[0] + 32).c_str()),
-            _("Error"), CClientUIInterface::OK | CClientUIInterface::MODAL);
```

Comment:
The authentication mechanism has been upgraded to not only use the old way, but also to accept an "authentication cookie", which is a temporary file generated on every run that contains a new password. Please check the authentication as a user and see if anything has changed. To me, it didn't look like anything has changed.

This change was necessary to get regtests to work, because they load the password from the cookie automatically.